Typescript - using withTheme is generating compilation errors when passing a function or a class as a parameter (2.1.0)

<!-- Issues that don't use this template are likely to be closed. -->
<!-- Please search the existing issues and read the documentation before opening an issue. -->

### Current behaviour

<!-- What's currently happening? -->
Using the withTheme function with a class or a function generates the following error : 

`error TS2349: Cannot invoke an expression whose type lacks a call signature. Type 'Pick<ThemingType<Theme, DeepPartial<Theme>>, "withTheme">' has no compatible call signatures.

export default withTheme(MyComponent);`

### Expected behaviour

<!-- What do you expect to happen? If there is an error, provide the complete error message with stack trace. -->

This should compile without problems (it was working fine in previous versions)

### Code sample

<!-- Provide a complete code sample that could be run to reproduce the issue, ideally on snack.expo.io or in a github repo. -->

Sample from help documents generates the error : 

`import * as React from 'react';
import { withTheme } from 'react-native-paper';

function MyComponent(props) {
  const { colors } = props.theme;
  return <Text style={{ color: colors.primary }}>Yo!</Text>;
}

export default withTheme(MyComponent);`

### Screenshots (if applicable)

<!-- Include screenshots if there's a bug in the layout. -->

### What have you tried

<!-- List down the steps you have tried to fix or identify the issue and links to any related issues you found. -->

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        | both
| react-native          | 0.57.2
| react-native-paper    | 2.1.0
| node                  |  10.11.0
| npm or yarn           | 6.4.1
| typescript | 3.1.1
| tsc | 3.0.3

--- a/typings/types.d.ts
+++ b/typings/types.d.ts
@@ -38,17 +38,13 @@ export type Theme = {
   };
 };
 
+type Theming = ThemingType<Theme, ThemeShape>;
+
 export type ThemeShape = DeepPartial<Theme>;
-export declare const withTheme: Pick<
-  ThemingType<Theme, ThemeShape>,
-  'withTheme'
->;
+export declare const withTheme: Theming['withTheme'];
 export declare const DefaultTheme: Theme;
 export declare const DarkTheme: Theme;
-export declare const ThemeProvider: Pick<
-  ThemingType<Theme, ThemeShape>,
-  'ThemeProvider'
->;
+export declare const ThemeProvider: Theming['ThemeProvider'];
 
 export declare class Provider extends React.Component<{
   children: React.ReactNode;

